### PR TITLE
233: Add a checkbox to the user edit page to set a user as a GlotPress admin

### LINKED ISSUE #233: Add UI to manage GP Administrators
Currently GlotPress has no UI to add or remove administrators.

### PR DESCRIPTION
**Pull request by toolstack**
Resolves #233.

### CODE REVIEW

File: gp-includes/misc.php [hunk 1]
```diff
@@ -325,3 +325,40 @@ function gp_gmt_strtotime( $string ) {
...
+
+function gp_personal_options( $user ) {
+	if ( ! current_user_can( 'manage_options' ) ) {
+		return;
+	}
+	
+?>
+<tr>
+<th scope="row"><?php _e( 'GlotPress Administrator', 'glotpress' ); ?></th>
+<td><fieldset><legend class="screen-reader-text"><span><?php _e('Toolbar') ?></span></legend>
```
Comment (ocean90) [inline gp-includes/misc.php]: Copy/paste error: Toolbar

Comment (toolstack) [inline gp-includes/misc.php]: Corrected.

File: gp-includes/misc.php [hunk 2]
```diff
@@ -325,3 +325,40 @@ function gp_gmt_strtotime( $string ) {
...
+
+<?php
+}
+
+function gp_personal_options_update( $user_id ) {
+	if ( ! current_user_can( 'manage_options' ) ) {
+		return;
+	}
+	
+	$current_gp_admin = GP::$permission->user_can( $user_id, 'admin' );
```
Comment (ocean90) [inline gp-includes/misc.php]: `$current_gp_admin` is confusing (it's not the current user), `$is_user_gp_admin` makes more sense IMO.

File: gp-includes/things/administrator-permission.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+class GP_Administrator_Permission extends GP_Permission {
+
+	var $table_basename = 'gp_permissions';
+	var $field_names = array( 'id', 'user_id', 'action', 'object_type', 'object_id' );
+	var $non_db_field_names = array();
+	var $non_updatable_attributes = array( 'id', );
+
+	function restrict_fields( $permission ) {
+		$permission->user_id_should_not_be('empty');
```
Comment (ocean90) [inline gp-includes/things/administrator-permission.php]: What about `$permission->action_should_not_be('empty');` or `$permission->object_type_should_be('empty');`?

File: gp-includes/misc.php [hunk 4]
```diff
@@ -325,3 +325,40 @@ function gp_gmt_strtotime( $string ) {
...
+	
+	if( array_key_exists( 'gp_administrator', $_POST ) && ! $current_gp_admin ) {
+		GP::$administrator_permission->create( array( 'user_id' => $user_id, 'action' => 'admin', 'object_type' => null ) );
+	}
+	
+	if( ! array_key_exists( 'gp_administrator', $_POST ) && $current_gp_admin ) {
```
Comment (ocean90) [inline gp-includes/misc.php]: A space is missing before the parens.

**Comment by ocean90** [comment]
I don't think that the checkbox fits into the "Personal Options" section. I'd create an new section only for GlotPress:

glotpress-section

That looks a bit more future-proof with #196 in mind.

Used code:

``` php
function gp_personal_options( $user ) {
    if ( ! current_user_can( 'manage_options' ) ) {
        return;
    }
    ?>
<h2><?php _e( 'GlotPress' ); ?></h2>
<table class="form-table">
    <tr>
        <th><?php _e( 'Admin', 'glotpress' ); ?></th>
        <td>
            <p>
                <label>
                    <input name="gp_administrator" type="checkbox" id="gp_administrator" value="1"<?php checked( GP::$permission->user_can( $user, 'admin' ) ); ?> />
                    <?php _e( 'Grant this user admin privileges for GlotPress.', 'glotpress' ); ?>
                </label>
            </p>
        </td>
    </tr>

    <?php do_action( 'gp_user_profile', $user ); ?>

</table>
    <?php
}
```

**Comment by toolstack** [comment]
I'm ok with a separate section, I'll update the PR.

**Pull request merged**